rfcs/0133-ipfs.md

```diff
@@ -0,0 +1,284 @@
+---
+feature: ipfs
+start-date: (fill me in with today's date, YYYY-MM-DD)
+author: John Ericsion (@Ericson2314) on behalf of [Obsidian Systems](https://obsidian.systems)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Integrate Nix with IPFS, in phases of increasing sophistication.
+This follows the work done and described in https://github.com/obsidiansystems/ipfs-nix-guide/ .
+
+# Motivation
+[motivation]: #motivation
+
+## Binary distribution
```

Comment:
IIUC, the phase one of this RFC is to use IPFS as a decentralized CDN for binary/source distributions. But I do think there still needs to be a benchmark on whether the transport is more efficient than `git fetch` of Git packfiles.

Reply:
`git fetch` and packfiles can do very wonderful and clever things, but acting as a CDN is not among them. I think this is an apples-and-oranges comparison without a clear answer benchmarking could resolve.

Reply:
One could have the mirrors in the form of Git repos, though this is not as automatic as running the IPFS daemon.

At the very least, for low-resource/remote systems, in addition to improving the download speed, I think we wouldn't want the package download to consume a lot of bandwidth overall? But then again, this optimization issue is implementation detail, and so is out of scope for this RFC.